crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1417,18 +1429,37 @@ impl<'db> TypeInferenceBuilder<'db> {
             .node_scope(NodeWithScopeRef::Function(function))
             .to_scope_id(self.db(), self.file());
 
-        let function_ty = Type::FunctionLiteral(FunctionType::new(
+        let mut inferred_ty = Type::FunctionLiteral(FunctionType::new(
             self.db(),
             &name.id,
             function_kind,
             body_scope,
-            decorator_tys.into_boxed_slice(),
+            function_decorators,
         ));
 
+        for (decorator_ty, decorator_node) in decorator_types_and_nodes.iter().rev() {
+            if let Type::FunctionLiteral(function) = decorator_ty {
+                if function.is_known(self.db(), KnownFunction::Overload) {
+                    continue;
+                }
+            }
```

Comment:
Is this needed? It looks like if we encounter `overload` above we don't push it onto `decorator_types_and_nodes`, so how would we then encounter it here?